### PR TITLE
Fixing FileNotFoundException occured while running sparkApp for query validation

### PR DESCRIPTION
**Pull request by sonalsagarwal**
## Changes proposed in this pull request
- Modifying test code to use SnappyTestUtils.writeToFile method for writing query result to csv.

- Changed default persistence mode to 'sync'.

## Patch testing
Ran ct.bt

### CODE REVIEW

File: dtests/src/test/scala/io/snappydata/hydra/TestUtil.scala [hunk 1]
```diff
@@ -179,7 +182,8 @@ object TestUtil {
       val logDir = log.listFiles.filter(_.getName.equals("snappyleader.log"))
       if (!logDir.isEmpty) {
         val leaderLogFile: File = logDir.iterator.next()
-        if (leaderLogFile.exists()) dest = dirString + File.separator + ".." + File.separator + ".." + File.separator + dirName
+        if (leaderLogFile.exists())
```
Comment (swatisawant) [inline dtests/src/test/scala/io/snappydata/hydra/TestUtil.scala]: Use the SnappyTestUtils.getTempDir() method instead of this one as both are same.

Comment (sonalsagarwal) [inline dtests/src/test/scala/io/snappydata/hydra/TestUtil.scala]: Done

**Comment by sonalsagarwal** [comment]
The methods are not exactly same.The logic is same, but the behavior is differnt, this has exception handling for few things, and the validation continues. Agree with duplication of code, but this can be done when we do refactoring of test code, to completely avoid duplicate classes.

**Comment by swatisawant** [comment]
Yes, you can take this up in next PR while refactoring the code. But, just make sure of modifying existing methods to support additional flags required for validation.

**Pull request merged**